homeworks/03-data-structure/src/binary-tree/binarySearchTree.test.ts

```diff
@@ -0,0 +1,41 @@
+import { BinarySearchTree } from './binarySearchTree';
+
+/*             43
+*             /  \
+*          40     46
+*         /  \   /  \
+*       10    2 1    5
+*              \
+*               67
+*/
+const givenSearchTree = new BinarySearchTree({
+  value: 43,
+  left: {
+    value: 40,
+    left: { value: 10, left: null, right: null },
+    right: {
+      value: 2,
+      left: null,
+      right: { value: 67, left: null, right: null },
+    },
+  },
+  right: {
+    value: 46,
+    left: { value: 1, left: null, right: null },
+    right: { value: 5, left: null, right: null },
+  },
+});
+
+describe('BinarySearchTree', () => {
```

Comment:
Why those methods should be tested once again and if should, how exhaustive should those be?

Reply:
My point of view here:
Tests should be independent in this case because `BinarySearchTree` is a special case of `BinaryTree` that can work only with numbers. Nobody can guarantee that numbers are covered in `BinaryTree` tests. `BinaryTree` tests can be changed, the logic of `BinaryTree` can be refactored, and better to double sure that `numbers` functionality is not affected.

Reply:
Ok, i got it. \
Only thing i can't imagine how can i check if inherited methods are covered with unit tests. This case is simple, but in big projects... 🤷 

 Added - https://github.com/vbarbasins/evo-ts-bootcamp/pull/3/commits/f5cb612033d9bb13e1d09dc3e278e513db56e8e5